### PR TITLE
initial code commit

### PR DESCRIPTION
**Pull request by vishalvvr**
#### Did:
- add all ansible collection source files
- add ansible-lint.yml & yml_lint_config files for bypassing  unnecessary linting errors
  and warning
- add GitHub action to test and push collection into ansible-galaxy

#### Note: 
Before merging this PR you need the following environment in place
- Add the ansible-galaxy API key into [GitHub secrets](https://github.com/distributed-system-analysis/pbench-ansible-collections/settings/secrets/actions) with the secret key name GALAXY_API_KEY.
- This codebase and GitHub action should be part of the "agent" branch, since we don't have it in our repo I am pushing it into this branch.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,46 @@
...
+- pbench_config_dest: the location within the pbench-agent installation directory where the config file is going to be copied (default: /opt/pbench-agent/config/)
+- pbench_config_files: a list of config files to be installed on an agent (default: '["pbench-agent.cfg"]').
+- pbench_key_url: the location where an ssh key can be obtained for sending results to a server (default: none).
+- pbench_key_dest: the location within the pbench-agent installation directory where the key is going to be copied. (default: /opt/pbench-agent/).
+
+Some of these are defaulted for convenience, but pbench_repo_url_prefix and pbench_key_url are specific to an environment and need to be specified locally (most likely in the inventory file).
+
+An example playbook can be obtained from
+
+    https://github.com/distributed-system-analysis/pbench/blob/master/agent/ansible/pbench_agent_install.yml
```
Comment (webbnh) [inline README.md:20]: References to "master" appear to redirect to "main", so we should presumably use the latter, here and elsewhere.

**Review by webbnh — COMMENTED** [review]
Are we sure that we want this stuff in its own repo?  It doesn't have any version/branch-specific dependencies?

**Comment by portante** [comment]
> Are we sure that we want this stuff in its own repo?

In the past I had understood that the GitHub integration with Ansible Galaxy required one namespace per repository.

But reading about Ansible Collections, that does not seem to be the case.

Can we get a good description for why this needs to be in a separate repository?

**Review by portante — CHANGES_REQUESTED** [review]
We need to be sure it is a requirement for this ansible galaxy collections work be located in a separate repository.

**Comment by vishalvvr** [comment]
> Are we sure that we want this stuff in its own repo? It doesn't have any version/branch-specific dependencies?

This ansible collection does not use anything from the pbench source tree, so we are good to separate the collection source from the existing tree.

**Comment by vishalvvr** [comment]
> > Are we sure that we want this stuff in its own repo?
> 
> In the past I had understood that the GitHub integration with Ansible Galaxy required one namespace per repository.
> 
> But reading about Ansible Collections, that does not seem to be the case.
>
yes, the ansible collection does not have any repo restriction, we just need to add the galaxy API key in GitHub secrets that's it.
   
> Can we get a good description for why this needs to be in a separate repository?

If we add GitHub action( to test, build and push to galaxy) to the existing pbench source tree, GitHub action won't be able to identify whether changes made are in the ansible collection source or pbench source.

After discussion with @ndokos we came to the conclusion that we should create a separate repo to simplify things.

But now I realize that the better approach would be to create a new branch(for the pbench-agent ansible collection) in the pbench repo itself.

File: galaxy.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+tags: [pbench]
+
+# Collections that this collection requires to be installed for it to be usable. The key of the dict is the
+# collection label 'namespace.name'. The value is a version range
+# L(specifiers,https://python-semanticversion.readthedocs.io/en/latest/#requirement-specification). Multiple version
+# range specifiers can be set and are separated by ','
+dependencies: {}
+
+# The URL of the originating SCM repository
+repository: http://github.com/distributed-system-analysis/pbench
```
Comment (webbnh) [inline galaxy.yml:49]: What is this used for?

File: galaxy.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+# The URL to any online docs
+documentation: https://distributed-system-analysis.github.io/pbench/doc/
+
+# The URL to the homepage of the collection/project
+homepage: http://github.com/distributed-system-analysis/pbench/agent
```
Comment (webbnh) [inline galaxy.yml:55]: This URL yields a 404.

File: galaxy.yml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+# The URL to the collection issue tracker
+issues: http://github.com/distributed-system-analysis/pbench/issues
```
Comment (webbnh) [inline galaxy.yml:58]: This URL yields a 404.

File: galaxy.yml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+# collection label 'namespace.name'. The value is a version range
+# L(specifiers,https://python-semanticversion.readthedocs.io/en/latest/#requirement-specification). Multiple version
+# range specifiers can be set and are separated by ','
+dependencies: {}
+
+# The URL of the originating SCM repository
+repository: http://github.com/distributed-system-analysis/pbench
+
+# The URL to any online docs
+documentation: https://distributed-system-analysis.github.io/pbench/doc/
```
Comment (webbnh) [inline galaxy.yml:52]: This URL yields a 404.

File: roles/pbench_agent_config/tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,19 @@
+---
+# pbench agent configuration
+- name: "pbench agent configuration - install config file(s)"
+  ansible.builtin.include_role:
+    name: pbench_agent_files_install
+  vars:
+    source: "{{ pbench_config_url }}"
```
Comment (webbnh) [inline roles/pbench_agent_config/tasks/main.yml:7]: Where/how is `pbench_config_url` defined, and what does it get defined to?  (Is it going to pull a file from the pbench Git repo?...if so, which _branch_ is it going to reference?)

**Review by webbnh — CHANGES_REQUESTED** [review]
I have nits and questions.

**Comment by webbnh** [comment]
> the better approach would be to create a new branch(for the pbench-agent ansible collection) in the pbench repo itself.

I doubt that that would be a better approach:  I think that creates confusion for anyone who is trying to access this content ("Why, when I get the `main` or default branch, can't I find the ansible collection??"), and it denies us the flexibility and power of using multiple branches for development.  That is, relative to putting it on the `main` branch or in its own repo, I think that's the worst of the three options.

> If we add GitHub action( to test, build and push to galaxy) to the existing pbench source tree, GitHub action won't be able to identify whether changes made are in the ansible collection source or pbench source.

Unless the proposed Action is very expensive (or there is some other downside to running it unnecessarily), I don't think this is a real problem.  We don't submit things to the `main` branch (or the release branches) very often, so I don't think unnecessary runs will be an issue.  On the other hand, if it _is_ an issue, it shouldn't be too hard to add some intelligence to the Action to skip the run if the changes aren't pertinent.

**Comment by portante** [comment]
We finalized that the ansible collections for the Pbench Agent are kept with the Pbench Agent code.

**Pull request closed without merging**